src/rules/maxLineLengthRule.ts

```diff
@@ -45,19 +73,31 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
+        const argument = this.ruleArguments[0];
+        const numberGreaterThan0 = (Number(argument) > 0);
+        const limitGreaterTHan0 = (argument instanceof Object && argument.limit > 0);
+        return super.isEnabled() && (numberGreaterThan0 || limitGreaterTHan0);
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        const argument = this.ruleArguments[0];
+        const options = {
+            ignorePattern: (typeof argument === "object") ? new RegExp(argument["ignore-pattern"]) : undefined,
+            limit: (typeof argument !== "object") ? parseInt(argument, 10)
+                : parseInt(argument.limit, 10),
+        };
+        return this.applyWithFunction(sourceFile, walk, options);
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
-    for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
-        }
+function walk(ctx: Lint.WalkContext<{limit: number; ignorePattern: RegExp | undefined}>) {
+    const limit = ctx.options.limit;
+    const ignorePattern = ctx.options.ignorePattern;
+    const lines = (ctx.sourceFile && ctx.sourceFile.text.split("\n") || []);
```

Comment:
please avoid this array by using `ctx.sourceFile.text.substr(line.pos, line.contentLength)` in the loop below